File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    N = len(clothes)
    wardrobe = defaultdict(int)
    
    # 같은 이름 옷 없으면 그냥 count로 해도 될거같은데?
    # -> 이 이유로 숫자값만 카운트
    for clothe in clothes:
        wardrobe[clothe[1]] += 1
    
    answer = 1
    
    for kind in wardrobe:
        # 입는 경우 + 안 입는 경우 모두 체크하기 위해 +1
        answer *= (wardrobe[kind] + 1)
        
    # 모두 안 입는 경우 제외
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    const wardrobe = {};
    const N = clothes.length;
    
    for (const clothe of clothes) {
        const kind = clothe[1];
        if (wardrobe[kind]) {
            wardrobe[kind] += 1;
        } else {
            wardrobe[kind] = 1;
        }
        
    }
    
    let answer = 1;
    
    for (const kind in wardrobe) {
        answer *= (wardrobe[kind] + 1);
    }
    
    return answer -1;
    
}